### PR TITLE
Refactor informer factories creation and initialization

### PR DESCRIPTION
**Pull request by derekbit**
- Refactor informer factories creation and initialization
- Filter pdb, daemonset and deployment by namespace

longhorn/longhorn#6954

### CODE REVIEW

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x

**Review by shuo-wu — COMMENTED** [review]
I am thinking if DaemonSet and Deployment can be filtered by namespace as well? The main blocker may be the rollout usage

**Comment by innobead** [comment]
> I am thinking if DaemonSet and Deployment can be filtered by namespace as well? The main blocker may be the rollout usage

Rollout should only care about longhorn-system namespace only, so it should be fine. @c3y1huang ?

File: datastore/datastore.go [hunk 1]
```diff
@@ -17,8 +16,8 @@ import (
 	storagelisters_v1 "k8s.io/client-go/listers/storage/v1"
 
 	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
-	lhinformers "github.com/longhorn/longhorn-manager/k8s/pkg/client/informers/externalversions"
 	lhlisters "github.com/longhorn/longhorn-manager/k8s/pkg/client/listers/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/util"
```
Comment (c3y1huang) [inline datastore/datastore.go:20]:
```suggestion
	"github.com/longhorn/longhorn-manager/util"

    lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
	lhlisters "github.com/longhorn/longhorn-manager/k8s/pkg/client/listers/longhorn/v1beta2"
```

File: util/client/client.go [hunk 2]
```diff
@@ -22,8 +21,12 @@ import (
 	"github.com/longhorn/longhorn-manager/datastore"
 	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
-	lhinformers "github.com/longhorn/longhorn-manager/k8s/pkg/client/informers/externalversions"
 	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/util"
+)
+
+const (
+	defaultResyncPeriod = 30 * time.Second
```
Comment (c3y1huang) [inline util/client/client.go:29]: Where is this used?

**Comment by shuo-wu** [comment]
OK we won't record the workload deployment/daemonset in the system backup. Namespace filtering seems fine. Let's wait for Chin-Ya's final confirmation

**Review by c3y1huang — COMMENTED** [review]
System backup/restore doesn't handle the application, should be fine.

File: util/client/client.go (hunk 2 above)
Comment (derekbit) [reply util/client/client.go:29]: Forgot to remove it. Thank you.

**Comment by derekbit** [comment]
@shuo-wu @c3y1huang @shuo-wu 
Just checked datastore's pod CRUD. We can filter pod by longhorn-system namespace? Right?

**Comment by shuo-wu** [comment]
> Just checked datastore's pod CRUD. We can filter pod by longhorn-system namespace? Right?

No. We will retrieve the workload status for volumes. And the workloads can be in any namespaces.

**Comment by derekbit** [comment]
> > Just checked datastore's pod CRUD. We can filter pod by longhorn-system namespace? Right?
> 
> No. We will retrieve the workload status for volumes. And the workloads can be in any namespaces.

Got it. Thank you.

Ref:
the namespace is from `pv.Spec.ClaimRef.Namespace`.
https://github.com/longhorn/longhorn-manager/blob/master/controller/kubernetes_pv_controller.go#L427

File: datastore/datastore.go (hunk 1 above)
Comment (derekbit) [reply datastore/datastore.go]: Updated. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**